src/main/java/com/sarality/db/query/SimpleQueryBuilder.java

```diff
@@ -46,6 +48,20 @@ public SimpleQueryBuilder withFilter(QueryClause clauseBuilder) {
     return this;
   }
 
+  //TODO (Satya) make the measureTypes an enum
+  public SimpleQueryBuilder withAggregate(String measureType, Column measureColumn) {
+    String columnName = measureType + "(" + measureColumn.getName() + ") AS " + measureType + "_" + measureColumn
+        .getName();
+    tableColumnList.add(columnName);
+    return this;
+  }
+
+  public SimpleQueryBuilder withGroupBy(Column dimension) {
```

Comment:
Do we want to make sure that if the Group is specified we specified at least one measure is also specified? This way you want leave the builder in a state where it doesn't make sense

groupBy(Column measure, AggregateFunction function, Column... groupByColumns)